### ISSUE TITLE
Better checkpoint or remove video from README?

### ISSUE THREAD

**Issue by avaer** [MineDojo/NitroGen#41]
I'll be upfront, the README seems extremely misleading with regard to this model's capabilities, as released.

The current checkpoint, `ng.pt`, is trained to inference a single 256x256 frame and predicts ~18 controller actions for a horizon of 0.3 seconds. That's it, unless I misunderstand (please correct me if that's the case).

If you actually try the model to do anything in practice you will see that it will mostly output random inputs, getting nowhere. I tried to run this on Sonic 3 + Knuckles emulation and although the model did know to press "right", it was not capable of jumping over a spring, given 10 minutes. Given that the the checkpoint understands only one 256x256 frame at a time, this makes sense.

You can see folks who have tried the model sharing similar experiences.

Developers who haven't tried the model are commenting in this repo with wild, unrealistic ideas of what this model checkpoint can do based on the flashy video in the README, none of which can be replicated.

Unless there a better checkpoint out there that does something remotely close to being able to replicate the gameplay seen at the top of the README, maybe the flashy video should be removed and the checkpoint capabilities described better so people have more realistic expectations of what the state of the current release is. Respectfully, this would save developers time.

**Comment by loicmagne** [comment]
Hi @avaer , thanks for your interest

This repository is an artifact of a research project that includes a large scale dataset, an evaluation API and a model. The full design and limitations of the model are explained in [the research report](https://arxiv.org/abs/2601.02427)

> NitroGen is limited to being a fast-reacting system-1 sensory model. It cannot plan over long horizons or follow language instructions; the model only reacts to the short context it sees. We develop NitroGen aiming for it to serve as a foundation for future generalist agent development, where post-training for language-following and reinforcement learning can be applied to enhance planning capabilities and improve success rates.

The current model is indeed a small 500M parameter diffusion transformer trained to predict actions from a single frame. It is not meant to play games end-to-end, self improve, or play completely unseen games. We hope it can serve as a framework to build more capable agents

I updated the README to include the limitations, since it seems like a few people skimmed past the paper. Thanks for your feedback!

**Comment by avaer** [comment]
Thank you @loicmagne.

Obviously a lot of folks are excited about further developments -- is the team planning on releasing either scaled up checkpoints with longer horizons (seems the code was built with this in mind), or training code so that this model can indeed serve as a "framework to build more capable agents"?

How do you envision folks building on top of the current repo? Or is that out of scope, and the only purpose of this repo is self-promotion for now?

I did notice that the license doesn't seem very friendly to "open source" usage so I'm curious how you hope this repo/research to be used in practical terms. Or is it just a "code dump" to generate hype? It would be nice if the expectations were clear.

**Comment by gloatoriginal** [comment]
I agree with @avaer this project in its current state is misleading. If it's for research purposes I lend to the question as well, who and what can use this to build off of? In it's current form it's less than useful, as it doesn't even show an example of how this should work on a base level.

For example I had it attempt to play cuphead starting from the menu, as well as starting from ingame, it never moved more than 1-2 steps and all of the output provided via terminal was 18 actions executed, and server.py saying None for all action_tensors, it seemed to just randomly send actions. In it's current form I can't imagine it's anything outside of a self-promotion ad to bring visuals to the repo/owner.

**Comment by loicmagne** [comment]
There's many things to build on top of this project:

- Train an IDM using the released dataset to label many gameplay videos. [These folks](https://si.inc/posts/fdm1/) have done it for computer use agent with 44k hours of labeled video data, so our dataset should be plenty enough
- Build a proper RL environment for games. Currently our `game_gym` only provides a `.step` function that executes actions and pause the game, but you could also use mods to reset player location/state, read the game state to create reward signal from enemy's HP, level completion etc. [This repo](https://si.inc/posts/fdm1/) does it for Dark Souls and Elden Ring but you could extend it to many games
- RL tune NitroGen to increase success rate on specific task. While not reliable, the current model can obtain rewards on various tasks in various game, which makes it a good starting point for RL fine-tuning. We were able to do self-imitation to increase the success rate against certain bosses in action RPGs
- Add language conditioning, long context planning, it's likely that there are better ways to leverage NitroGen's dataset for that

Many open questions and I believe our open release provides tools to tackle them that weren't available before. If you're not satisfied by this release feel free to use what other labs have open-sourced =)
